tests/unit/frameworks/pytorch/kwinner2d_local_test.py

```diff
@@ -0,0 +1,159 @@
+        """
+        x = self.x[0:1]
+        n, c, h, w = x.shape
+        kw = KWinners2dLocal(
+            percent_on=0.5,  # k=2
+            channels=c,
+            k_inference_factor=1.0,
+            boost_strength=0.0,
+            duty_cycle_period=1000
+        )
+        kw.train(mode=False)
+
+        expected = torch.zeros_like(x)
+
+        expected[0, [2, 3], 0, 0] = x[0, [2, 3], 0, 0]
+        expected[0, [1, 3], 0, 1] = x[0, [1, 3], 0, 1]
+        expected[0, [0, 1], 1, 0] = x[0, [0, 1], 1, 0]
+        expected[0, [2, 3], 1, 1] = x[0, [2, 3], 1, 1]
+
+        result = kw(x)
+        self.assertEqual(result.shape, expected.shape)
+
+        num_correct = (result == expected).sum()
+        self.assertEqual(num_correct, result.reshape(-1).size()[0])
+
+    def test_k_winners2d_two(self):
+        """
+        Equal duty cycle, boost_strength=0, percent_on=0.5, batch size=2
+        """
+        x = self.x[0:2]
+        n, c, h, w = x.shape
+        kw = KWinners2dLocal(
+            percent_on=0.5,  # k=2
+            channels=c,
+            k_inference_factor=1.0,
+            boost_strength=0.0,
+            duty_cycle_period=1000
+        )
+        kw.train(mode=False)
+        expected = torch.zeros_like(x)
+
+        expected[0, [2, 3], 0, 0] = x[0, [2, 3], 0, 0]
+        expected[0, [1, 3], 0, 1] = x[0, [1, 3], 0, 1]
+        expected[0, [0, 1], 1, 0] = x[0, [0, 1], 1, 0]
+        expected[0, [2, 3], 1, 1] = x[0, [2, 3], 1, 1]
+
+        expected[1, [2, 3], 0, 0] = x[1, [2, 3], 0, 0]
+        expected[1, [1, 3], 0, 1] = x[1, [1, 3], 0, 1]
+        expected[1, [0, 3], 1, 0] = x[1, [0, 3], 1, 0]
+        expected[1, [0, 2], 1, 1] = x[1, [0, 2], 1, 1]
+
+        result = kw(x)
+        self.assertEqual(result.shape, expected.shape)
+
+        num_correct = (result == expected).sum()
+        self.assertEqual(num_correct, result.reshape(-1).size()[0])
+
+    def test_k_winners2d_train(self):
+        """
+        Test training
+        Changing duty cycle, boost_strength=1, percent_on=0.5, batch size=2
+        """
+        x = self.x[0:2]
+        n, c, h, w = x.shape
+        kw = KWinners2dLocal(
+            percent_on=0.5,
+            channels=c,
+            boost_strength=1.0,
+            duty_cycle_period=10,
+        )
+
+        kw.train(mode=True)
+        expected = torch.zeros_like(x)
+        expected[0, [2, 3], 0, 0] = x[0, [2, 3], 0, 0]
+        expected[0, [1, 3], 0, 1] = x[0, [1, 3], 0, 1]
+        expected[0, [0, 1], 1, 0] = x[0, [0, 1], 1, 0]
+        expected[0, [0, 1], 1, 1] = x[0, [0, 1], 1, 1]
+
+        expected[1, [2, 3], 0, 0] = x[1, [2, 3], 0, 0]
+        expected[1, [1, 3], 0, 1] = x[1, [1, 3], 0, 1]
+        expected[1, [0, 3], 1, 0] = x[1, [0, 3], 1, 0]
+        expected[1, [0, 2], 1, 1] = x[1, [0, 2], 1, 1]
+
+        result = kw(x)
+        result = kw(x)
+        self.assertTrue(result.eq(expected).all())
+
+        expected_boosted = expected.clone()
```

Comment:
This confused me at first. I would add the comment:
``
        # Expectation due to boosting after the fourth training step
``